runtime/Java/src/org/antlr/v4/runtime/Parser.java

```diff
@@ -594,35 +593,21 @@ public Token consume() {
 	}
 
 	/** How to create a token leaf node associated with a parent.
-	 *  Typically, the terminal node to create is not a function of the parent
-	 *  but this method must still set the parent pointer of the terminal node
-	 *  returned. I would prefer having {@link ParserRuleContext#addAnyChild(ParseTree)}
-	 *  set the parent pointer, but the parent pointer is implementation dependent
-	 *  and currently there is no setParent() in {@link TerminalNode} (and can't
-	 *  add method in Java 1.7 without breaking backward compatibility).
+	 *  Typically, the terminal node to create is not a function of the parent.
 	 *
 	 * @since 4.6.1
 	 */
 	public TerminalNode createTerminalNode(ParserRuleContext parent, Token t) {
-		TerminalNodeImpl node = new TerminalNodeImpl(t);
-		node.parent = parent;
-		return node;
+		return new TerminalNodeImpl(t);
```

Comment:
Hmm, in what sense? The parent is set by the caller, so why pass it into this method?